src/service/video-manager-impl.js

```diff
@@ -437,6 +556,216 @@ class VideoEntry {
+        0, this.initialHeight_,
+        DOCK_SCALE, 1
+    ) + ')';
+    this.internalStyle_.borderRadius = mapRange(inViewportHeight,
+        this.initialHeight_, 0,
+        0, 6
+    ) + 'px';
+    // Shadow position
+    const sPos = '0px 0px 20px 0px';
+    // Shadow color
+    const sCol = '0, 0, 0'; // rgb black
+    // Shadow opacity
+    const sOpac = mapRange(inViewportHeight,
+        this.initialHeight_, 0,
+        0, 0.3
+    );
+    this.internalStyle_.boxShadow = sPos + ' rgba(' + sCol + ',' + sOpac + ')';
+    this.internalStyle_.left = mapRange(inViewportHeight,
+        this.initialHeight_, 0,
+        this.initialLeft_, 20
+    ) + 'px';
+
+    // Different behavior based on whether the video got minimized
+    // from the top or the bottom
+    if (this.minimizePosition_ == MinimizePositions.TOP) {
+
+      this.internalStyle_.top = mapRange(inViewportHeight,
+          this.initialHeight_, 0,
+          0, 20
+      ) + 'px';
+      this.internalStyle_.bottom = 'auto';
+      this.internalStyle_.transformOrigin = 'top left';
+
+    } else {
+
+      // Revert back to inline viewing
+
+      this.internalStyle_.bottom = mapRange(inViewportHeight,
+          this.initialHeight_, 0,
+          0, 20
+      ) + 'px';
+      this.internalStyle_.top = 'auto';
+      this.internalStyle_.transformOrigin = 'bottom left';
+    }
+
+    // TODO(@wassim) Make minimized video draggable
+  }
+
+  /**
+   * Applies final transformations to the docked video to assert that the final
+   * position and scale of the docked video are correct (in case user scrolls
+   * too fast for startDocking_ to kick in)
+   *
+   * NOTE(@wassgha) : won't be needed if PositionObserver returned the element's
+   * position when it goes out of view.
+   * @private
+   */
+  endDocking_() {
+    // Hide the controls.
+    this.video.hideControls();
+    this.internalClasses_.add(DOCK_CLASS);
+    this.internalStyle_.transform = 'scale(' + DOCK_SCALE + ')';
+    this.internalStyle_.borderRadius = '6px';
+    this.internalStyle_.left = '20px';
+    this.internalStyle_.boxShadow = '0px 0px 20px 0px rgba(0, 0, 0, 0.3)';
+    if (this.minimizePosition_ == MinimizePositions.BOTTOM) {
+      this.minimizePosition_ = MinimizePositions.TOP;
+      this.internalStyle_.bottom = '20px';
+      this.internalStyle_.top = 'auto';
+      this.internalStyle_.transformOrigin = 'bottom left';
+    } else {
+      this.minimizePosition_ = MinimizePositions.BOTTOM;
+      this.internalStyle_.top = '20px';
+      this.internalStyle_.bottom = 'auto';
+      this.internalStyle_.transformOrigin = 'top left';
+    }
+  }
+
+  /**
+   * Restores styling of the video to make it go back to its original inline
+   * position.
+   *
+   * @private
+   */
+  unDockVideo_() {
+    // Restore the video inline
+    this.internalClasses_.remove(DOCK_CLASS);
+    this.internalElement_.setAttribute('style', '');
+    if (!this.hasAutoplay || this.userInteractedWithAutoPlay_) {
```

Comment:
We are docking videos that have autoplay but are manually playing though, right?